File: app/src/main/java/com/yamikrish/app/mvpapplication/users/detail/DetailActivity.java
package com.yamikrish.app.mvpapplication.users.detail;
/*
 * Created by Yamini on 24/5/18.
 */

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.yamikrish.app.mvpapplication.R;

import com.yamikrish.app.mvpapplication.utils.Utils;

public class DetailActivity extends AppCompatActivity {

    DetailFragment fragment;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if(fragment == null){
            fragment = DetailFragment.newInstance();
            Utils.addFragmentToActivity(getSupportFragmentManager(), fragment, R.id.container);
        }
    }
}

File: app/src/main/java/com/yamikrish/app/navigationDemo/ui/detail/DetailActivity.kt
package com.yamikrish.app.navigationDemo.ui.detail

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import com.yamikrish.app.navigationDemo.R
import com.yamikrish.app.navigationDemo.model.Post
import kotlinx.android.synthetic.main.detail_page.*

/* 
 * Created by Yamini on 15/5/18.
 */

class DetailActivity : AppCompatActivity() {

    private lateinit var viewModel: DetailViewModel
    lateinit var post: Post

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.detail_page)
        val id = DetailActivityArgs.fromBundle(intent.extras).id
        val factory = CustomViewModelFactory(id)

        val toolbar: Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)

        supportActionBar?.let {
            it.title = getString(R.string.post_detail)
            it.setDisplayShowHomeEnabled(true)
            it.setDisplayHomeAsUpEnabled(true)
        }

        viewModel = ViewModelProviders.of(this, factory).get(DetailViewModel::class.java)

        viewModel.fetchPostById().observe(this, object : Observer<Post> {
            override fun onChanged(t: Post?) {
                postTitle.text = t?.title
                description.text = t?.body
            }

        })
    }

    class CustomViewModelFactory(private val test: Int) : ViewModelProvider.NewInstanceFactory() {
        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
            return DetailViewModel(test) as T
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        finish()
        return  true;
    }
}
